client/electron/index.ts

```diff
@@ -529,53 +529,59 @@ function main() {
   // If the function encounters an error, it throws an Error that can be parsed by the `PlatformError`.
   ipcMain.handle(
     'outline-ipc-invoke-method',
-    (_, method: string, params: string): Promise<string> =>
-      invokeMethod(method, params)
-  );
-
-  // Connects to a proxy server specified by a config.
-  //
-  // If any issues occur, an Error will be thrown, which you can try-catch around
-  // `ipcRenderer.invoke`. But you should avoid depending on the specific error type.
-  // Instead, you should use its message property (which would probably be a JSON representation
-  // of a PlatformError). See https://github.com/electron/electron/issues/24427.
-  //
-  // TODO: refactor channel name and namespace to a constant
-  ipcMain.handle(
-    'outline-ipc-start-proxying',
-    async (_, request: StartRequestJson): Promise<void> => {
-      // TODO: Rather than first disconnecting, implement a more efficient switchover (as well as
-      //       being faster, this would help prevent traffic leaks - the Cordova clients already do
-      //       this).
-      if (currentTunnel) {
-        console.log('disconnecting from current server...');
-        currentTunnel.disconnect();
-        await currentTunnel.onceDisconnected;
-      }
+    async (_, method: string, params: string): Promise<string> => {
+      // TODO(fortuna): move all other IPCs here.
+      switch (method) {
+        case 'StartProxying': {
```

Comment:
*nit* Do we really need the `{` here?

Reply:
Yes, otherwise we can't declare local variables exclusive to this case.